app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/indicators/TrackerAnalyticsRepository.kt

```diff
@@ -0,0 +1,218 @@
+package org.dhis2.usescases.teiDashboard.dashboardfragments.indicators
+
+import dhis2.org.analytics.charts.Charts
+import io.reactivex.Flowable
+import io.reactivex.Observable
+import io.reactivex.functions.Function3
+import org.dhis2.data.analytics.AnalyticsModel
+import org.dhis2.data.analytics.ChartModel
+import org.dhis2.data.analytics.IndicatorModel
+import org.dhis2.data.analytics.LOCATION_FEEDBACK_WIDGET
+import org.dhis2.data.analytics.LOCATION_INDICATOR_WIDGET
+import org.dhis2.data.analytics.SectionTitle
+import org.dhis2.data.forms.dataentry.RuleEngineRepository
+import org.dhis2.data.tuples.Pair
+import org.dhis2.data.tuples.Trio
+import org.dhis2.utils.DhisTextUtils
+import org.dhis2.utils.Result
+import org.dhis2.utils.resources.ResourceManager
+import org.hisp.dhis.android.core.D2
+import org.hisp.dhis.android.core.arch.helpers.UidsHelper
+import org.hisp.dhis.android.core.program.ProgramIndicator
+import org.hisp.dhis.android.core.program.ProgramRuleActionType
+import org.hisp.dhis.rules.models.RuleActionDisplayKeyValuePair
+import org.hisp.dhis.rules.models.RuleActionDisplayText
+import org.hisp.dhis.rules.models.RuleEffect
+import timber.log.Timber
+
+class TrackerAnalyticsRepository(
```

Comment:
Do you think that we could add tests for this class?

Reply:
done!